client/src/pages/admin/Users.tsx

```diff
@@ -1,7 +1,550 @@
+          >
+            <X className="h-4 w-4" />
+          </Button>
+        </div>
+      );
+    });
+
+    return activeFilters.length > 0 ? (
+      <div className="flex flex-wrap gap-2 mt-4">{activeFilters}</div>
+    ) : null;
+  };
+
+  return (
+    <div className="p-4">
+      <h1 className="text-2xl font-semibold mb-4">Users Management</h1>
+
+      <div className="mb-6 p-4 border rounded shadow-sm">
+        <div className="flex justify-between items-center mb-3">
+          <h2 className="text-lg font-medium">Filters</h2>
+          <Button
+            variant="outline"
+            onClick={clearFilters}
+            className="flex items-center gap-2 text-black border-black hover:bg-black/5"
+            disabled={
+              !filters.name &&
+              !filters.education_level.length &&
+              !filters.major.length &&
+              !filters.role.length &&
+              !filters.paid.length
+            }
+          >
+            <X className="h-4 w-4" />
+            Clear All Filters
+          </Button>
+        </div>
+
+        <div className="grid grid-cols-1 md:grid-cols-2 lg:grid-cols-5 gap-4">
+          <div>
+            <label className="block text-sm mb-1">Name</label>
+            <input
+              type="text"
+              className="w-full p-2 border rounded"
+              value={nameSearch}
+              onChange={handleNameSearchChange}
+              placeholder="Search by name"
+            />
+          </div>
+
+          <FilterDropdown
+            label="Education Level"
+            options={educationLevelOptions}
+            value={filters.education_level}
+            onChange={(values) => handleFilterChange('education_level', values)}
+          />
+
+          <FilterDropdown
+            label="Major"
+            options={majorOptions}
+            value={filters.major}
+            onChange={(values) => handleFilterChange('major', values)}
+          />
+
+          <FilterDropdown
+            label="Role"
+            options={roleOptions}
+            value={filters.role}
+            onChange={(values) => handleFilterChange('role', values)}
+          />
+
+          <FilterDropdown
+            label="Membership Term"
+            options={paidOptions}
+            value={filters.paid}
+            onChange={(values) => handleFilterChange('paid', values)}
+          />
+        </div>
+
+        {renderFilterChips()}
+      </div>
+
+      {isLoading ? (
+        <Spinner />
+      ) : error ? (
+        <div className="text-red-500 p-4">Error loading users data</div>
+      ) : users.length > 0 ? (
+        <>
+          <UsersTable users={users} />
+          <DataTablePagination
+            currentPage={currentPage}
+            totalPages={totalPages}
+            itemsPerPage={itemsPerPage}
+            onPageChange={setCurrentPage}
+            onItemsPerPageChange={handleItemsPerPageChange}
+            itemsPerPageOptions={[10, 20, 50]}
+          />
```

Comment:
I was expecting the pagination template to have children prop, which would display the contents of that page. so in this case, `UsersTable` would be the template's child component. Something like the suggestion.
```suggestion
          <DataTablePagination
            currentPage={currentPage}
            totalPages={totalPages}
            itemsPerPage={itemsPerPage}
            onPageChange={setCurrentPage}
            onItemsPerPageChange={handleItemsPerPageChange}
            itemsPerPageOptions={[10, 20, 50]}
          >
          <UsersTable users={users} />
          </DataTablePagination>
```

But your way works. you don't have to change it.